internal/replacements/replacements.go

```diff
@@ -246,7 +247,14 @@ func processGroupNeedsRemovalForPod(cluster *fdbv1beta2.FoundationDBCluster, pod
 		}
 	}
 
-	return false, nil
+	desiredPod, err := internal.GetPod(cluster, processGroupStatus)
+	if err != nil {
+		return false, err
+	}
+	// TODO deprecated builtin k8s features edited securityContext automatically, and it doesn't seem outlandish that someone's cluster
```

Comment:
We have similar cases in the statements above where some other controllers are adding additional information in the spec. That's the reason we have this line at some checks:


```go
if pod.ObjectMeta.Annotations[fdbv1beta2.LastSpecKey] != specHash {
    // replace ..
}
```

In this case the replacement will only be triggered if the applied specHash is different from the last applied spec hash. This can still result in replacements that shouldn't be a replacement, e.g. if the spec hash changes because of another field but now the operator detects the mismatch of the security context and triggers a replacement. But as far as I know, this is the best we can do right now.

Reply:
My feeling is that adding annotations for securityContext spec change might not be warranted, but that adding a feature flag to turn it off could be worthwhile, but I know that flag bloat is very real

Reply:
discussed offline: added a feature flag